scrapper.py

```diff
@@ -44,20 +77,111 @@ def get_search_urls(self):
+            self.article.author = "NOT FOUND"
+        # DATE
+        raw_date = article_bs.find('div', class_='time').text
+        months_dict = {'января': '01',
+                       'февраля': '02',
+                       'марта': '03',
+                       'апреля': '04',
+                       'мая': '05',
+                       'июня': '06',
+                       'июля': '07',
+                       'августа': '08',
+                       'сентября': '09',
+                       'октября': '10',
+                       'ноября': '11',
+                       'декабря': '12',
+                       }
+        for month in months_dict:
+            if month in raw_date:
+                raw_date = raw_date.replace(month, months_dict[month])
+            elif month.capitalize() in raw_date:
+                raw_date = raw_date.replace(month.capitalize(), months_dict[month])
+        self.article.date = datetime.strptime(raw_date, '%d %m %Y %H:%M')
+        # TOPICS
+        topics_bs = article_bs.find('div', class_='entry__tags').find_all('a')
+        for topic_bs in topics_bs:
+            self.article.topics.append(topic_bs.text[1:])
+
+    def _fill_article_with_text(self, article_bs):
+        text_of_article = ''
+        texts = article_bs.find('article', class_='entry__body').find_all('p')
+        for text_part in texts:
+            text_of_article += text_part.text
+        self.article.text = text_of_article
+
+    def parse(self):
+        response = requests.get(url=self.article_url)
+        article_bs = BeautifulSoup(response.text, 'lxml')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path = Path(base_path)
+    if path.exists():
+        shutil.rmtree(path)
+    path.mkdir(parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as file:
+        config = json.load(file)
+
+    for url in config["seed_urls"]:
+        if HTTP_PATTERN not in url:
+            raise IncorrectURLError
+
+    seed_urls = config["seed_urls"]
+    max_articles = config["total_articles_to_find_and_parse"]
+
+    if not seed_urls:
+        raise IncorrectURLError
+
+    if not isinstance(max_articles, int) or max_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if max_articles > 200:
+        raise NumberOfArticlesOutOfRangeError
+
+    return seed_urls, max_articles
 
 
 if __name__ == '__main__':
-    # YOUR CODE HERE
-    pass
+    test_seed_urls, test_max_articles = validate_config(CRAWLER_CONFIG_PATH)
+    prepare_environment(ASSETS_PATH)
+
+    print('craw')
+    crawler = Crawler(test_seed_urls, test_max_articles)
+    crawler.find_articles()
+
+    print('pars')
+    ARTICLES_ID = 1
```

Comment:
in python, we usually use capital letters to denote constants 
you are changing the value of `ARTICLES_ID` after each iteration, so it is not exactly constant 
I would recommend to use `enumerate()` for declaring a variable that would serve as an ascending index

Reply:
why can't i stick to this method if it works?

Reply:
It definitely does work, however, there are certain [conventions](https://realpython.com/python-pep8/#naming-styles) that make your code more readable
So it is not so much about functioning, but about making it easier for others to orient in your code

Reply:
done with enumerate